celery-java/src/main/java/com/geneea/celery/Celery.java

```diff
@@ -55,15 +55,23 @@
     private Celery(final String brokerUri,
                    @Nullable final String queue,
                    @Nullable final String backendUri,
-                   @Nullable final ExecutorService executor) {
+                   @Nullable final ExecutorService executor,
+                   @Nullable final boolean isPriQueue,
+                   @Nullable final int maxPriority) {
```

Comment:
Could `maxPriority` be an `Optional<Integer>`? That way you wouldn't need `isPriQueue` parameter.